plugins/bitbucket-cloud-common/scripts/adjust-models.js

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+
+const modelsFile = `${__dirname}/../src/models/index.ts`;
+
+let models = fs.readFileSync(modelsFile).toString();
+
+// the generator creates ${DefinitionName}AllOf interfaces for the "allOf" statements with a type definition
+// (as extension to another references one) as well as sub-interfaces or enums.
+// Those are duplicates of the interface ${DefinitionName}.
+const matches = models.matchAll(
+  /(interface|enum|const|type) (\w+AllOf\w*)[ \n]+[^{=]*[{=]/gm,
+);
```

Comment:
Gotta say that I would be a little bit more at ease with this entire thing if it was done with AST operations instead 😅, something like `ts-morph`. Feel it's fine to merge as is but it's a potential improvement.

Reply:
I have no idea what ts-morph is. Fairly new to typescript, etc. In general, I know AST operations though.

In general, I wished all of this was not necessary :-/
but the generator is not good enough at certain parts and the spec itself on others or even a little buggy.